src/frontend/src/binder/select.rs

```diff
@@ -769,14 +768,67 @@ impl Binder {
             .unzip()
     }
 
-    fn bind_distinct_on(&mut self, distinct: Distinct) -> Result<BoundDistinct> {
+    /// Bind `DISTINCT` clause in a [`Select`].
+    /// Note that for `DISTINCT ON`, each expression is interpreted in the same way as `ORDER BY`
+    /// expression, which means it will be bound in the following order:
+    ///
+    /// * as an output-column name
+    /// * as an index (from 1) of an output column
+    /// * as an arbitrary expression
```

Comment:
Strictly speaking (tested in PG), it's arbitrary expression except for non-integer constants...

```
> select * from t order by 'a';
non-integer constant in ORDER BY
LINE 1: select * from t order by 'a'
                                 ^
-- But constexpr is ok...
select * from t order by length('a');
```
🤣

Reply:
It seems not hard to fix this (just adding a `Expr::Value(_)` match scrutine). So (just slightly) better to fix it.

Reply:
In the latter case, will it order by the constant or the column corresponding to the constant as ordinal number?

Reply:
by constant. so it's quite a silly usage. 😟